src/ducks/recurrence/service.js

```diff
@@ -19,20 +19,31 @@ export const NB_DAYS_LOOKBACK = 100
 
 /**
  * Fetch transactions from 3 months back if there is already recurrences,
- * otherwise all transactions
+ * otherwise all transactions.
+ * We filter out transactions already associated with a recurrence as we don't
+ * want to change it.
+ *
  * @param {array} recurrences - Hydrated bundle of recurrences
  * @returns transactions
  */
 const makeQueryForTransactions = recurrences => {
   if (recurrences.length === 0) {
-    return Q(TRANSACTION_DOCTYPE)
+    // We still filter out transactions that were explicitely marked as not
+    // recurrent.
+    return Q(TRANSACTION_DOCTYPE).partialFilter({
+      'relationships.recurrence': { $exists: false }
+    })
```

Comment:
You may want to specify a `.limit(1000)`, otherwise, the pagination will be 100 by default, resulting in ~10x more queries